### PR TITLE
Minor update to releasing docs

### PR DESCRIPTION
**Pull request by bsipocz**
As we suggest elsewhere not to have a local branch named ``master``.

Edit: I've also added minor update to the helpers release. This PR should remain open until the last minute, so more updates can be added while we do the release.

TODO:

* [x] - add description of twine, currently it's only mentioned for the helpers
* [ ] - revise suggestions from #5120. We already update the release docs after the 1.3 release, so I expect most of it is either covered already or outdated (the numbering is definitely outdated)

EDIT: Other suggestions based on 2.0 experience

 * [ ] - point 12&13 could be moved to the bottom as there is no development going on on a bugfix branch anyway, so both the version and changelog can be dealt with once everything is ready and we're just waiting for the conda packages to being built
 * [ ] - if we keep building the wheels as part of the release, point 25 needs to be done before the wheels as they need to know about the new version tag

### CODE REVIEW

**Comment by bsipocz** [comment]
Should include https://github.com/astropy/astropy/issues/4434, too.

File: docs/development/releasing.rst [hunk 1]
```diff
@@ -630,6 +630,9 @@ applies both for regular release *and* release candidates are the same
       $ git checkout <maintenance branch name>
       $ git merge --no-ff tmp-release-v<version>
       $ git tag -s "v<version>" -m "Tagging v<version>"
+      $ git clean -dfx
+      $ umask 0022
```
Comment (eteq) [inline docs/development/releasing.rst]: I think it's worth adding just a sentence or two in the text explaining why you're doing this - it's not really obvious from this what the point is.

Comment (bsipocz) [inline docs/development/releasing.rst]: it's already mentioned above in the main part, isn't that enough?

**Comment by bsipocz** [comment]
@eteq - I'll need to revise the upload part and add twine, so I'm not aiming to merge this before 2.0.0, but in the next bugfix.

**Comment by bsipocz** [comment]
@eteq - the tmp branching in the helpers release instruction raised some issues for v1.3 (https://github.com/astropy/astropy-tools/issues/32) so this time I hasn't followed it. Should I revise it in the docs, or keep it around a bit more?

**Comment by bsipocz** [comment]
Actually I very much dislike to use the third party download and upload to pypi script for the wheels, so  would rather update the instructions to download all wheel locally, and then upload them along with the sdist using the same twine command. What do you think?

**Comment by eteq** [comment]
I think this is good as it stands @bsipocz.  I think one could argue that the wheels build should be treated separately to simplify this document, but it's probably better to discuss that in a separate PR with a concrete case - either way all this content is needed, so I'll go ahead and mrege.

**Comment by bsipocz** [comment]
Hmm, I should have added the WIP label, the TODO wasn't yet fully finished. Anyway, that can be done in a new PR at some point :)

**Comment by eteq** [comment]
Oh, my bad, I misunderstood and thought your last few commits had done this, @bsipocz .  Sorry about that!

**Comment by bsipocz** [comment]
No worries. 
Actually merging this cleared up already addressed issues as well as making the release procedure easier as I can simply follow it from the rendered docs page rather than from WIP PRs.

The unaddressed points now moved over to #5120, so we now only have one place that relates to the further updates which is very good.

**Pull request merged**